libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -1810,14 +1819,18 @@ uint16_t AP_Mission::get_landing_sequence_start()
 bool AP_Mission::jump_to_landing_sequence(void)
 {
     uint16_t land_idx = get_landing_sequence_start();
-    if (land_idx != 0 && set_current_cmd(land_idx)) {
+    if (land_idx != 0 && set_current_cmd(land_idx, true)) {
 
         //if the mission has ended it has to be restarted
         if (state() == AP_Mission::MISSION_STOPPED) {
             resume();
         }
 
         gcs().send_text(MAV_SEVERITY_INFO, "Landing sequence start");
+
+        // set flag
+        _flags.jumped_to_landing = true;
```

Comment:
You need to set this flag to true if you also just fly through a `DO_LAND_START` in the mission, otherwise the general case where the vehicle finished the rest and is flying the planned landing loses the protection.

You'd also want to flag this as true if you are flying a NAV_LAND or NAV_VTOL_LAND item, as clearly you'd want to continue.

Reply:
It now sets the flag true when flies through DO_LAND_START.

The NAV_LAND and NAV_VTOL_LAND scenarios are already handled by events.cpp